clientv3/balancer.go

```diff
@@ -175,37 +174,30 @@ func (b *simpleBalancer) Get(ctx context.Context, opts grpc.BalancerGetOptions)
 		b.mu.RLock()
 		closed := b.closed
 		addr = b.pinAddr
-		upEps := len(b.upEps)
 		b.mu.RUnlock()
 		if closed {
 			return grpc.Address{Addr: ""}, nil, grpc.ErrClientConnClosing
 		}
 
-		if upEps == 0 {
+		if addr == "" {
 			return grpc.Address{Addr: ""}, nil, ErrNoAddrAvilable
 		}
 		return grpc.Address{Addr: addr}, func() {}, nil
 	}
 
-	for {
```

Comment:
I think this still needs to loop, otherwise there's a race:
1. Connection is up, b.upc is closed
2. Get fetches ch, falls through select since it's closed
3. Connection goes down, pinAddr is set to "", `b.upc` set to new channel 
4. Get fetches pinAddr, gets "", returns ClientClosing, instead of waiting until next connection is up.